### PR TITLE
Pack rat monster, the missing ratkin link

### PR DESCRIPTION
**Pull request by Doitallornot123**
#### Summary
Content "Adds new humanoid "pack rat" monster"

#### Purpose of change
I noticed there was a "missing link" in the rat kin monster group. Difficulty-wise, the ratkin group goes from dog sized rat, straight into magical telekenetic half human half rat hybrid, or even a half rat, sword wielding "mausketeer". This monster was made with the intention of bridging the gap.

#### Describe the solution
essentially a slightly buffed up tunnel rat. still leagues below the mausketeer in terms of danger. pack rat has the gimmick of dropping a wide variety of "shinies" or trash from the pack rat's "treasure" stuffed mouth. (almost all trash or mundane shiny items)

#### Describe alternatives you've considered
i considered a more wacky approach, with smoke bombs and insulting voicelines, but it felt too silly for the game's setting.

#### Testing
booted the game, made a new character and loaded into game proper with no errors, spawned pack rat using the debug menu, monster and monster drops functioned as intended

#### Additional context
my first attempt at adding a monster to the game, i appreciate any advice.
Screenshot 2022-08-31 173510

### CODE REVIEW

**Comment by Maleclypse** [comment]
It doesn't have to carry it in it's mouth, it could make sense for something that size and desire to hoard things to were a utility vest or makeshift equivalent.

File: data/json/monsters/rodentkin.json [hunk 1]
```diff
@@ -141,6 +141,45 @@
     "flags": [ "SEES", "SMELLS", "HEARS", "WARM", "SWIMS", "PATH_AVOID_DANGER_1", "CAN_DIG" ],
     "dissect": "dissect_rat_sample_small"
   },
+  {
+    "id": "mon_pack_rat",
+    "type": "MONSTER",
+    "name": { "str": "pack rat" },
+    "description": "This humanoid rodent stands over five feet tall, it pokes and bites at anything shiny, hastily shoving anything of interest into its mouth",
```
Comment (Fris0uman) [inline data/json/monsters/rodentkin.json:148]: Keeping stuff in their cheeks is a hamster thing, not a rat thing as far as I know.

File: data/json/monsters/rodentkin.json [hunk 2]
```diff
@@ -141,6 +141,45 @@
...
+    "melee_dice": 4,
+    "melee_dice_sides": 4,
+    "melee_damage": [ { "damage_type": "cut", "amount": 5 } ],
+    "dodge": 3,
+    "armor_bash": 5,
+    "armor_cut": 6,
+    "armor_bullet": 2,
+    "anger_triggers": [ "PLAYER_WEAK", "PLAYER_CLOSE", "FRIEND_ATTACKED", "STALK" ],
+    "path_settings": { "max_dist": 50, "avoid_traps": true, "avoid_sharp": true },
+    "death_drops": "ratkin_death_drops_pack-rat",
```
Comment (Fris0uman) [inline data/json/monsters/rodentkin.json:167]:
```suggestion
    "death_drops": "ratkin_death_drops_pack_rat",
```
Maybe name the id `ratkin_death_drops_pack_rat`

File: data/json/monsterdrops/rodentkin.json [hunk 3]
```diff
@@ -41,6 +41,18 @@
       { "group": "accesories_personal_unisex", "damage": [ 0, 2 ], "prob": 45 }
     ]
   },
+  {
+    "type": "item_group",
+    "subtype": "distribution",
+    "//": "contains shiny drops from a pack rat's mouth",
+    "id": "ratkin_death_drops_pack-rat",
```
Comment (Fris0uman) [inline data/json/monsterdrops/rodentkin.json:48]:
```suggestion
    "id": "ratkin_death_drops_pack_rat",
```

**Comment by Doitallornot123** [comment]
> It doesn't have to carry it in it's mouth, it could make sense for something that size and desire to hoard things to were a utility vest or makeshift equivalent.

While I think that's true, my reason for making the monster originally was to bridge the evolutionary (and difficulty) gap between what is essentially a rat dog ("mon_tunnel_rat"), and a sword wielding, properly dressed, seemingly fully intelligent rat man mutant ("mon_mausketeer"). I think the behavior of stuffing shiny objects into its mouth like a savage fits the ape-like level of intelligence I had in my head.

File: data/json/monsters/rodentkin.json (hunk 1 above)
Comment (Doitallornot123) [reply data/json/monsters/rodentkin.json:148]: That's true. Normal IRL pack rats do carry single objects in their hands or mouth to transport back to their nests though, unless I'm misinformed. I thought it wouldn't be too much of a stretch to imagine that a significantly larger packrat could carry more than one coin/piece of trash.

Comment (Fris0uman) [inline data/json/monsters/rodentkin.json:148]:
```suggestion
    "description": "This humanoid rodent stands over five feet tall, it pokes and bites at anything shiny, hastily shoving anything of interest into its mouth.",
```

**Pull request merged**